src/Commands/BaseCommand.php

```diff
@@ -129,13 +132,53 @@ protected function promptForMissingArguments(InputInterface $input, OutputInterf
         );
     }
 
-    protected function getModuleModel($name)
+    protected function getModuleModel($name): Module
     {
         return $name instanceof \Nwidart\Modules\Module
             ? $name
             : $this->laravel['modules']->findOrFail($name);
     }
 
+    /**
+     * Read the statuses file defined in config('modules.activators.file.statuses-file')
+     * and return enabled module names preserving the order in that file.
+     * Falls back to repository ordering if file is missing or invalid.
+     */
+    private function getModulesOrderedByStatusesFile(): array
+    {
+        $statusesFile = $this->laravel['config']->get('modules.activators.file.statuses-file');
+        $statusesFile = is_string($statusesFile) && $statusesFile !== ''
+            ? $statusesFile
+            : base_path('modules_statuses.json');
+
+        $sort = $this->option('direction') ?? 'asc';
+
+        if (! is_string($statusesFile) || ! file_exists($statusesFile)) {
+            return array_keys($this->laravel['modules']->getOrdered($sort));
+        }
+
+        $json = @file_get_contents($statusesFile);
+        $data = is_string($json) ? json_decode($json, true) : null;
+
+        if (! is_array($data)) {
+            return array_keys($this->laravel['modules']->getOrdered($sort));
+        }
+
+        $modules = [];
+        foreach ($data as $name => $enabled) {
+            if ($enabled === true || $enabled === 1) {
+                $modules[] = $name; // Keep StudlyCase names as defined
+            }
+        }
+
+        // Fallback if no enabled modules found
+        if (empty($modules)) {
+            return array_keys($this->laravel['modules']->getOrdered($sort));
+        }
```

Comment:
This part is incorrect. If no modules are enabled, the command shouldn’t run across all modules. Running migrations when no modules are enabled doesn’t make sense and could cause unintended side effects.